oss-internship-2020/guetzli/guetzli_transaction.cc

```diff
@@ -0,0 +1,120 @@
+#include "guetzli_transaction.h"
+
+#include <iostream>
+#include <memory>
+
+namespace guetzli {
+namespace sandbox {
+
+absl::Status GuetzliTransaction::Init() {
+  // Close remote fd if transaction is repeated
+  if (in_fd_.GetRemoteFd() != -1) {
+    SAPI_RETURN_IF_ERROR(in_fd_.CloseRemoteFd(sandbox()->GetRpcChannel()));
+  }
+  if (out_fd_.GetRemoteFd() != -1) {
+    SAPI_RETURN_IF_ERROR(out_fd_.CloseRemoteFd(sandbox()->GetRpcChannel()));
+  }
+
+  // Reposition back to the beginning of file
+  if (lseek(in_fd_.GetValue(), 0, SEEK_CUR) != 0) {
```

Comment:
no need to first check the position, just set it to 0.
it might be easier/clearer to just open the files inside the transaction.
e.g. what happens now if transaction fails in the middle of writting to out_fd?